[Julius Gambe] iP

juliusgambe opened the pull request:
# Gobble Gobble

Initial Pull Request of **Gobble** **Gobble** 🏁.  This project is based on [Duke](https://dev.java/duke/).

## Changes
- Add Ui
- Add Storage
- Add Parser
- Add Command
- Add Task

## Type of Tasks
1. TodoTask
2. DeadlineTask
3. EventTask

## Type of Commands
1. AddCommand
2. DeleteCommand
3. ExitCommand
4. ListCommand
5. MarkCommand
6. UnmarkCommand

## Review Checklist
- [ ] My code follows the **style guidelines**
- [ ] My code follows **OOP**
- [ ] My code satisfies a _minimal_ amount of tests critical to the project
- [ ] My code has **javadocs** as its documentation

## Code Starting Point
Code starts at `Duke.java`

```java
 public void run() {
        ui.showWelcomeMessage();
        Scanner scanner = new Scanner(System.in);
        while (true) {
            String userInput = scanner.nextLine().trim();
            try {
                Command command = Parser.parse(userInput);
                command.execute(tasks, ui, storage);
                if (command.isExit()) {
                    break;
                }

            } catch (DukeException e) {
                ui.showError(e.getMessage());
            }

        }
        scanner.close();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

juliusgambe committed:
feat: rename, greet, exit

juliusgambe committed:
feat: echo

juliusgambe committed:
feat: add, list

juliusgambe committed:
fix: added whitespace

juliusgambe committed:
add: task class

juliusgambe committed:
feat: mark as done

juliusgambe committed:
fix: duke encapsulation

juliusgambe committed:
add: task toString

juliusgambe committed:
add: ToDoTask

juliusgambe committed:
add: EventTask

juliusgambe committed:
add: DeadlineTask

juliusgambe committed:
feat: ToDos, Events, Deadlines

juliusgambe committed:
test: add input and expected

juliusgambe committed:
add: duke exceptions

juliusgambe committed:
add: delete

juliusgambe committed:
add: command enum

juliusgambe committed:
Fix enum error

juliusgambe committed:
Add save

juliusgambe committed:
Add dates

juliusgambe committed:
Merge pull request #1 from juliusgambe/branch-Level-7

Add save

juliusgambe committed:
Merge pull request #2 from juliusgambe/branch-Level-8

Add dates and times

juliusgambe committed:
Fix duplicate functions

juliusgambe committed:
Add Parser.java

juliusgambe committed:
Add Ui.java

juliusgambe committed:
Add Storage.java

juliusgambe committed:
Add TaskList.java

juliusgambe committed:
Add commands

juliusgambe committed:
Handle date formatter exception

juliusgambe committed:
Add Command.java

juliusgambe committed:
Refactor Duke.java

juliusgambe committed:
Merge pull request #3 from juliusgambe/branch-A-MoreOOP

Refractor to OOP

juliusgambe committed:
Create duke, task and command package

juliusgambe committed:
Merge pull request #4 from juliusgambe/refractor-packages

Create duke, task and command packages

juliusgambe committed:
Merge pull request #5 from juliusgambe/add-gradle-support

Add gradle support

juliusgambe committed:
Add DeadlineTest.java

juliusgambe committed:
Add EventTest.java

juliusgambe committed:
Fix parseTodo

juliusgambe committed:
Add ParserTest.java

juliusgambe committed:
Refractor TaskList.java and add TaskListTest.java

juliusgambe committed:
Refractor tasks and Parser.java

juliusgambe committed:
Refractor ListCommand.java

juliusgambe committed:
Add test

juliusgambe committed:
Refactor Ui print list

juliusgambe committed:
Merge pull request #6 from juliusgambe/test-junit

Test junit

juliusgambe committed:
Add shadow jar properties

juliusgambe committed:
Add find

juliusgambe committed:
Add TaskType.java

juliusgambe committed:
Refactor Storage.java

juliusgambe committed:
Add storage javadocs

juliusgambe committed:
Refactor files

juliusgambe committed:
Merge pull request #7 from juliusgambe/branch-A-CodingStandard

Branch a coding standard

juliusgambe committed:
Merge branch 'master' into branch-Level-9

juliusgambe committed:
Add javadocs

juliusgambe committed:
Merge branch 'master' into branch-A-JavaDoc

juliusgambe committed:
Merge pull request #9 from juliusgambe/branch-A-JavaDoc

Add javadocs

juliusgambe committed:
Fix TaskList.java

juliusgambe committed:
Merge branch 'master' into branch-Level-9

juliusgambe committed:
Merge pull request #8 from juliusgambe/branch-Level-9

Add find

juliusgambe committed:
Fix TaskList.java

juliusgambe committed:
Fix Ui.java

juliusgambe committed:
Add javadocs to Ui.java

juliusgambe committed:
Add checkstyle

juliusgambe committed:
Fix styling errors

juliusgambe committed:
Add javafx

juliusgambe committed:
Refractor ui

juliusgambe committed:
Refractor fxml

juliusgambe committed:
Fix styling errors

juliusgambe committed:
Merge pull request #10 from juliusgambe/branch-Level-10

Branch level 10

antonTan96 commented on src/main/java/duke/enums/TaskType.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Represents a TaskType enum that contains the types of tasks.
+ */
+public enum TaskType {
+    TODO,
+    DEADLINE,
+    EVENT
+}
```
I like how you seperated the task types into another file :D

antonTan96 replied:
I like how you seperated the task types into another file :D

antonTan96 commented on src/main/java/duke/task/DeadlineTask.java:
```diff
@@ -0,0 +1,43 @@
+public class DeadlineTask extends Task {
+    private final LocalDate by;
+
+    /**
+     * Constructor for DeadlineTask.
+     *
+     * @param description of the task.
+     * @param by          deadline of the task.
+     */
```
I'm not sure if this follows the coding standard.
I would much prefer `@param description The description of the task`

antonTan96 reviewed (commented):
Looks good to me! No glaring issues and the coding standard is followed.

jianyangg commented on src/main/java/duke/ui/GobbleMessage.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Controller for a Message Bubble in the GUI.
+ */
+public class GobbleMessage extends HBox {
```
Wow! What an interesting and creative class name. 🤯

jianyangg commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,36 @@
+    public GobbleMessage execute(TaskList taskList, Storage storage) {
+        taskList.unmarkTask(index);
+        storage.saveListToDisk(taskList.getTasks());
+        return GobbleMessage.getDukeDialog("Nice! I've unmarked this task as undone:\n" + taskList.getTask(index),
```
Could this GobbleMessage be gobbling up too many characters (i.e., past the limit of 120 chars)?

jianyangg commented on src/main/java/duke/ui/MainWindow.java:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * Adds messages to the dialog container.
+     *
+     * @param messages messages to be added.
```
Similar to what Anton mentioned earlier and referring to CS2103T Java Conventions, we'll have to capitalise our descriptive text after the params. See this snippet from the textbook: 

`@param zone Zone of position.`

jianyangg reviewed (commented):
LGTM! Just a few nits to fix such as JavaDocs conventions and line character limit. Very creative work and elegant code nonetheless!

juliusgambe committed:
Add intro message

juliusgambe committed:
Add image

juliusgambe committed:
Merge pull request #11 from juliusgambe/branch-Profile-Image

Add image

juliusgambe committed:
Change main class name shadow jar

juliusgambe committed:
Add assertions and fixes

juliusgambe committed:
Refactor date parser

juliusgambe committed:
Fix GobbleMessage

juliusgambe committed:
Fix CodeQuality

Fix CodeQuality

Current code quality does not comply with coding standards.

Update code to comply with coding standards
* unused methods
* javadocs sentence casing
* keep it simple
* remove repetitons

juliusgambe committed:
Merge pull request #12 from juliusgambe/branch-A-CodeQuality

Branch a code quality

juliusgambe committed:
Merge branch 'master' into branch-A-Assertions

juliusgambe committed:
Merge pull request #13 from juliusgambe/branch-A-Assertions

Add assertions and fixes

juliusgambe committed:
Add help command

Add help command

No help command present.

Without help command, user may not know what commands are available.

Let's,
* add a help command that lists all the available commands and format

juliusgambe committed:
Fix javadocs

juliusgambe committed:
Rename TodoCommand to TaskCommand

juliusgambe committed:
Add window title

juliusgambe committed:
Fix help and find command

juliusgambe committed:
Remove throws

juliusgambe committed:
Remove unused import statement

juliusgambe committed:
Add StorageTest.java

juliusgambe committed:
Remove DeadlineTest exception

juliusgambe committed:
Remove trailing space

juliusgambe committed:
Change FindCommand message

juliusgambe committed:
Add Ui.png and change README.md

juliusgambe committed:
Remove duke.txt
